### ISSUE TITLE
file_cache_path setting

### ISSUE THREAD

**Issue by CMeza99** [chef/chef#3890]
When using a setting such as: 

``` ruby
file_cache_path '~/.chef/cache'
```

chef-client and chef-solo return:

```
[2015-09-07T14:17:06-07:00] INFO: Forking chef instance to converge...
[2015-09-07T14:17:06-07:00] ERROR: No such file or directory @ rb_sysopen - ~/.chef/cache/chef-client-running.pid
[2015-09-07T14:17:06-07:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

It does work when the full path is given. But this is unexpected as settings such as cookbook_path and data_bag_path will accept paths with ~.

chef 12.4.1

**Comment by lamont-granquist** [comment]
Seems related to another one we had recently about ".." not working in cookbook_path.

We should fix all of these and make sure they all work with ".." and "~".

**Comment by lamont-granquist** [comment]
Seems related to #2809 and #3079

**Comment by vsingh-msys** [comment]
Fixed #3890 via #9042
